DirectProgramming/C++SYCL_FPGA/ReferenceDesigns/svd/README.md

```diff
@@ -0,0 +1,354 @@
+| svd_testbench_tool.hpp  | Host code                       | Helper functions that is used in the test bench of the SVD demonstration to check for correctness.                                                                                                   |
+| print_matrix.hpp       | Host code                       | Helper functions to print matrices.                                                                                                                                                                  |
+| golden_pca.hpp         | Host code                       | A CPU reference PCA design that is used to calculate reference eigen values for the test bench.                                                                                                      |
+| svd.hpp                | Host code with some device code | Contains wrapper function `SingularValueDecomposition` that launches individual kernels of the SVD design.                                                                                           |
+| memory_transfers.hpp   | Device code                     | Contains kernel implementations of `MatrixReadFromDDRTo2PipesByBlocks` , `MatrixReadPipeToDDR` and `VectorReadPipeToDDR`. These kernels transfers data between DDR and streaming interfaces (pipes).|
+| post_process.hpp       | Device code                     | Contains kernel implementation of `USVFromEigens`.                                                                                                                                                   |
+### Input covariance matrix computation
+The covariance computation in this design is the same as used in the [PCA](../pca/README.md) reference design except without standardization.
+
+The covariance of input A is equal to the transpose of A multiplied by A:
+```math
+Cov(A) = A^{T}A
+```
+Therefore this kernel performs a matrix multiplication by blocks, as described in the PCA reference design.
+
+### Eigenvalues and Eigenvectors computation
+We are reusing the `fpga_linalg::StreamingEigen` design from the PCA sample to compute eigenvalues and eigenvectors of the input covariance matrix. These eigenvalues and eigenvectors are going to be used to construct the final outputs of SVD.
+
+This kernel works only with input of rank sufficient matrix (all columns are linearly independent). It produces an output `rank_deficent_flag` to indicate if the input matrix is not linearly independent. If the flag is set to 1, the result of the SVD is known to be incorrect.
+
+The source code for the the eigenvalues and eigenvectors computation kernel can be found in `streaming_eigen.hpp` in the shared [include](../../include) directory. An detailed explanation of the algorithm of computing eigenvalues can be found [here](../pca/README.md#eigen-values-and-eigen-vectors-computation) in the PCA sample.
+
+### Construct Output from Eigenvalues and Eigenvectors
```

Comment:
(I assume) that the computations in this section are all found in post_process.hpp; please state explicitly state this so the reader knows where to look.